neurons/validators/validator.py

```diff
@@ -110,40 +106,49 @@ def cross_validate(self, axon, node, start_block_height, last_block_height):
         return result, response_time
 
 
-    def get_reward(self, response: DiscoveryOutput, ip_per_hotkey=None, run_id_per_hotkey=None, miner_distribution=None):
-        output: DiscoveryOutput = response.output
-        network = output.metadata.network
-        start_block_height = output.start_block_height
-        last_block_height = output.block_height
-        axon_ip = response.axon.ip
-        hot_key = response.axon.hotkey
-        bt.logging.info(f"🔄 Processing response for {hot_key}@{axon_ip}")
-
-        multiple_ips = ip_per_hotkey[axon_ip] > MAX_MULTIPLE_IPS
-        multiple_run_ids = run_id_per_hotkey[hot_key] > MAX_MULTIPLE_RUN_ID
-
-        cross_validation_result, response_time = self.cross_validate(response.axon, self.nodes[network], start_block_height, last_block_height)
-
-        if cross_validation_result is None:
+    def get_reward(self, response: Discovery, ip_per_hotkey=None, run_id_per_hotkey=None, miner_distribution=None):
+        try:
+            if not is_discovery_response_valid(response):
+                bt.logging.debug(f'Discovery Response invalid {response}')
+                return None
+            
+            output: DiscoveryOutput = response.output
+            network = output.metadata.network
+            start_block_height = output.start_block_height
+            last_block_height = output.block_height
+            axon_ip = response.axon.ip
+            hot_key = response.axon.hotkey
+
+            bt.logging.info(f"🔄 Processing response for {hot_key}@{axon_ip}")
+
+            multiple_ips = ip_per_hotkey[axon_ip] > MAX_MULTIPLE_IPS
+            multiple_run_ids = run_id_per_hotkey[hot_key] > MAX_MULTIPLE_RUN_ID
+            cross_validation_result, response_time = self.cross_validate(response.axon, self.nodes[network], start_block_height, last_block_height)
```

Comment:
https://github.com/blockchain-insights/blockchain-data-subnet/blob/4bf3a589eb242be16309b184d1332a43e2a5985f/neurons/validators/validator.py#L88  please check also range check and implement it too for 2.1